### ISSUE TITLE
Clippy does not run on proc macro crates

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#2526]
```bash
#!/bin/sh

cargo new repro
echo >repro/src/lib.rs '
#[allow(dead_code)]
fn f() {
    let _x = 3.14;
}
'

echo >>repro/Cargo.toml '
[lib]
proc-macro = true
'

cargo clippy --manifest-path=repro/Cargo.toml
```

The same script without `proc-macro = true` does catch the `approx_constant` lint.
